commit db6ef53164cc03d9e696f21664da0c092617b4bd
Author: Shawn M Moore <shawn@bestpractical.com>
Date:   Fri Aug 18 17:47:41 2017 +0000

    Avoid spuriously unchecking all recipient checkboxes
    
    This 4.4.2 regression, introduced in
    f0a7105da8bc8f00a54bcdb5bfd48d788bbb888b and
    4b11cf37360ad5084f6428c62553ccc6858313a7, could happen if you caused a
    second refresh while the first one was still running. serializeArray()
    doesn't include disabled form fields, so what the AJAX request reported
    to the server made it look like the user had explicitly unchecked
    recipients.
    
    Fixes: I#33027

diff --git a/share/html/Ticket/Update.html b/share/html/Ticket/Update.html
index 075940306..8b4ba94c2 100644
--- a/share/html/Ticket/Update.html
+++ b/share/html/Ticket/Update.html
@@ -217,8 +217,18 @@ jQuery( function() {
        jQuery('#recipients div.titlebox-content').addClass('refreshing');
        jQuery('#previewscrips div.titlebox-content').addClass('refreshing');
 
+       /* temporarily re-enable the checkboxes so they will be included in
+          payload */
+       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', false);
+       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', false);
+
+       var payload = jQuery('form[name=TicketUpdate]').serializeArray();
+
+       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', true);
+       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', true);
+
        jQuery('#recipients div.titlebox-content').load( '<% RT->Config->Get('WebPath')%>/Helpers/ShowSimplifiedRecipients',
-           jQuery('form[name=TicketUpdate]').serializeArray(),
+           payload,
            function() {
                jQuery('#recipients div.titlebox-content').removeClass('refreshing');
                var txn_send_field = jQuery("#recipients input[name=TxnSendMailTo]");
@@ -232,7 +242,7 @@ jQuery( function() {
        );
 
        jQuery('#previewscrips div.titlebox-content').load( '<% RT->Config->Get('WebPath')%>/Helpers/PreviewScrips',
-           jQuery('form[name=TicketUpdate]').serializeArray(),
+           payload,
            function() {
                jQuery('#previewscrips div.titlebox-content').removeClass('refreshing');
                var txn_send_field = jQuery("#previewscrips input[name=TxnSendMailTo]");
@@ -244,9 +254,6 @@ jQuery( function() {
                }
            }
        );
-
-       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', true);
-       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', true);
    };
    updateScrips();
 
commit b68fdcbf6006922e60afa3da247472f4b7a96bbc
Author: sunnavy <sunnavy@bestpractical.com>
Date:   Fri Mar 9 21:57:11 2018 +0800

    Always enable recipient checkboxes
    
    The "disable" behavior is introduced in f0a7105d and 4b11cf37, which is
    mainly to indicate that recipients checkboxes are being updated and
    shouldn't be changed. It had a js execute order issue, which was fixed
    in db6ef531. But from real user reports, there is still possibility that
    checkboxes are not enabled in time, which causes RT to not send
    notifications.
    
    It introduces so much mess but gains so less, not mentioning that the
    0.3 opacity is already an indicator of "watch out, the section is being
    updated".

diff --git a/share/html/Ticket/Update.html b/share/html/Ticket/Update.html
index 8b4ba94c29..f90a89a0d7 100644
--- a/share/html/Ticket/Update.html
+++ b/share/html/Ticket/Update.html
@@ -217,16 +217,8 @@ jQuery( function() {
        jQuery('#recipients div.titlebox-content').addClass('refreshing');
        jQuery('#previewscrips div.titlebox-content').addClass('refreshing');
 
-       /* temporarily re-enable the checkboxes so they will be included in
-          payload */
-       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', false);
-       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', false);
-
        var payload = jQuery('form[name=TicketUpdate]').serializeArray();
 
-       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', true);
-       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', true);
-
        jQuery('#recipients div.titlebox-content').load( '<% RT->Config->Get('WebPath')%>/Helpers/ShowSimplifiedRecipients',
            payload,
            function() {
@@ -266,11 +258,6 @@ jQuery( function() {
 
    jQuery("#ticket-update-metadata :input, #UpdateCc, #UpdateBcc").change( updateScrips );
 
-   jQuery("body").on("submit", "form", function () {
-       jQuery("#recipients input[name=TxnSendMailToAll], #recipients input[name=TxnSendMailTo]").attr('disabled', false);
-       jQuery("#previewscrips input[name=TxnSendMailToAll], #previewscrips input[name=TxnSendMailTo]").attr('disabled', false);
-   });
-
 });
 </script>
 % }
